implement strndupa which musl does not provide.
copied from a patchset at the openembedded project:
https://git.openembedded.org/openembedded-core/plain/meta/recipes-core/systemd/systemd/0005-src-basic-missing.h-check-for-missing-strndupa.patch

--- python-systemd-234/systemd/util.c.orig
+++ python-systemd-234/systemd/util.c
@@ -32,6 +32,15 @@
 #include <unistd.h>
 #include <net/if.h>
 
+#define strndupa(s, n) \
+  ({ \
+    const char *__old = (s); \
+    size_t __len = strnlen(__old, (n)); \
+    char *__new = (char *)alloca(__len + 1); \
+    __new[__len] = '\0'; \
+    (char *)memcpy(__new, __old, __len); \
+  })
+
 #include "util.h"
 
 int safe_atou(const char *s, unsigned *ret_u) {
